website/server/controllers/api-v3/auth.js

```diff
@@ -532,6 +532,9 @@ api.updateEmail = {
     let validationErrors = req.validationErrors();
     if (validationErrors) throw validationErrors;
 
+    let userSameEmail = await User.findOne({'auth.local.email': req.body.newEmail}).exec();
```

Comment:
this should be `await User.findOne({'auth.local.email': req.body.newEmail}).select({_id: 1}.lean().exec(); ` so that only the _id field is loaded and not the entire user